common/network-common/src/main/java/org/apache/spark/network/client/StreamInterceptor.java

```diff
@@ -28,17 +28,17 @@
  * An interceptor that is registered with the frame decoder to feed stream data to a
  * callback.
  */
-class StreamInterceptor implements TransportFrameDecoder.Interceptor {
+class StreamInterceptor<T> implements TransportFrameDecoder.Interceptor {
```

Comment:
Before the request it is decided whether the use stream or chunk fetch. See https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/client/TransportResponseHandler.java#L75
and 
https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/client/TransportResponseHandler.java#L93

If we just Decode StreamResponse instead of ChunkFetchSuccess it wont work out of the box.